File: zig/src/World/ClusterInfo.zig
//! TODO RE-EVALULATE
//! TODO DOCUMENTATION

const ClusterInfo = @This();

name: []const u8,
ip: []const u8,
port: u16,

pub fn init(name: []const u8, ip: []const u8, port: u16) ClusterInfo {
    return ClusterInfo{
        .name = name,
        .ip = ip,
        .port = port,
    };
}

File: rust/src/world/cluster_info.rs
//! TODO RE-EVALULATE
//! TODO DOCUMENTATION

use std::net::Ipv4Addr;

pub struct ClusterInfo {
    name: String,
    ip: Ipv4Addr,
    port: u16,
}

impl ClusterInfo {
    pub fn new(name: String, ip: Ipv4Addr, port: u16) -> Self {
        ClusterInfo { name, ip, port }
    }
}
